opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -0,0 +1,255 @@
+ * abandoned cart emails with Smaily templates.
+ *
+ * @package smaily_for_opencart
+ * @author Smaily
+ * @license GPL-3.0+
+ * @copyright 2019 Smaily
+ *
+ * Plugin Name: Smaily for Opencart
+ * Description: Smaily email marketing and automation extension plugin for Opencart.
+ * Version: 1.1.3
+ * License: GPL3
+ * Author: Smaily
+ * Author URI: https://smaily.com/
+ *
+ * Smaily for Opencart is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * Smaily for Opencart is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Smaily for Opencart. If not, see <http://www.gnu.org/licenses/>.
+ */
+class ControllerModuleSmailyForOpencart extends Controller {
+    private $error = array();
+
+    public function index() {
+        // Add language file.
+        $this->load->language('module/smaily_for_opencart');
+        // Smaily admin page model.
+        $this->load->model('smailyforopencart/admin');
+        // Add heading title.
+        $this->document->setTitle($this->language->get('heading_title'));
+        
+        // Text fields
+        $this->data['heading_title'] = $this->language->get('heading_title');
+        $this->data['text_edit'] = $this->language->get('text_edit');
+
+        // Subdomain.
+        $this->data['subdomain_title'] = $this->language->get('entry_subdomain_title');
+        $this->data['subdomain_placeholder'] = $this->language->get('placeholder_subdomain');
+        // Username.
+        $this->data['username_title'] = $this->language->get('entry_username_title');
+        $this->data['username_placeholder'] = $this->language->get('placeholder_username');
+        // Password.
+        $this->data['password_title'] = $this->language->get('entry_password_title');
+        $this->data['password_placeholder'] = $this->language->get('placeholder_password');
+        // Credentials validated status.
+        $this->data['validated'] = $this->model_smailyforopencart_admin->getSettingValue('smaily_for_opencart_validated');
+        // Validate button.
+        $this->data['button_validate'] = $this->language->get('button_validate');
+        $this->data['validate_title'] = $this->language->get('validate_title');
+
+        // Small texts.
+        $this->data['small_subdomain'] = $this->language->get('small_subdomain');
+        $this->data['small_password'] = $this->language->get('small_password');
+        $this->data['token'] = $this->session->data['token'];
+
+        // Display warning.
+        if (isset($this->error['warning'])) {
```

Comment:
Variable `$this->error` keys are never set, and `$this->data['error_...']` is always empty. This code is redundant and can be removed.